logstash-core/spec/logstash/pipeline_dlq_commit_spec.rb

```diff
@@ -68,6 +68,7 @@ def close() end
 
   after(:each) do
     FileUtils.remove_entry pipeline_settings["path.dead_letter_queue"]
+    pipeline_settings_obj.reset
```

Comment:
could we instead ensure the object use isn't global? so in [here](https://github.com/guyboertje/logstash/blob/639a39e45071f713f7c985f7d33e8f53796fe0ef/logstash-core/spec/logstash/pipeline_dlq_commit_spec.rb#L39)

`let(:pipeline_settings_obj) { LogStash::SETTINGS }`
maybe do instead:  
`let(:pipeline_settings_obj) { LogStash::SETTINGS.clone }` ?